### PR TITLE
[MINOR][SQL][DOCS] Improve unix_timestamp's scaladoc (and typo hunting)

### PR DESCRIPTION
**Pull request by jaceklaskowski**
## What changes were proposed in this pull request?

* Docs are consistent (across different `unix_timestamp` variants and their internal expressions)
* typo hunting

## How was this patch tested?

local build

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -2487,19 +2487,21 @@ object functions {
...
   def current_timestamp(): Column = withExpr { CurrentTimestamp() }
 
+  // scalastyle:off line.size.limit
   /**
    * Converts a date/timestamp/string to a value of string in the format specified by the date
    * format given by the second argument.
    *
    * A pattern could be for instance `dd.MM.yyyy` and could return a string like '18.03.1993'. All
-   * pattern letters of `java.text.SimpleDateFormat` can be used.
+   * pattern letters of [[https://docs.oracle.com/javase/8/docs/api/java/text/SimpleDateFormat.html java.text.SimpleDateFormat]] can be used.
```
Comment (srowen) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: We don't generally link to external javadoc, and we've had problems creating a normal javadoc link to the class in many instances. For consistency, I wouldn't bother with this.

**Comment by jaceklaskowski** [comment]
Are the errors (that led to `fails to generate documentation`) after my change? Look very weird to me.

```
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/scheduler/BlacklistTracker.java:117: error: ExecutorAllocationClient is not public in org.apache.spark; cannot be accessed from outside package
[error]   public   BlacklistTracker (org.apache.spark.scheduler.LiveListenerBus listenerBus, org.apache.spark.SparkConf conf, scala.Option<org.apache.spark.ExecutorAllocationClient> allocationClient, org.apache.spark.util.Clock clock)  { throw new RuntimeException(); }
[error]                                                                                                                                                    ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/scheduler/BlacklistTracker.java:118: error: ExecutorAllocationClient is not public in org.apache.spark; cannot be accessed from outside package
[error]   public   BlacklistTracker (org.apache.spark.SparkContext sc, scala.Option<org.apache.spark.ExecutorAllocationClient> allocationClient)  { throw new RuntimeException(); }
[error]                                                                                             ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/SparkConf.java:133: error: ConfigReader is not public in org.apache.spark.internal.config; cannot be accessed from outside package
[error]   private  org.apache.spark.internal.config.ConfigReader reader ()  { throw new RuntimeException(); }
[error]                                            ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/SparkConf.java:138: error: ConfigEntry is not public in org.apache.spark.internal.config; cannot be accessed from outside package
[error]    <T extends java.lang.Object> org.apache.spark.SparkConf set (org.apache.spark.internal.config.ConfigEntry<T> entry, T value)  { throw new RuntimeException(); }
[error]                                                                                                 ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/SparkConf.java:139: error: OptionalConfigEntry is not public in org.apache.spark.internal.config; cannot be accessed from outside package
[error]    <T extends java.lang.Object> org.apache.spark.SparkConf set (org.apache.spark.internal.config.OptionalConfigEntry<T> entry, T value)  { throw new RuntimeException(); }
[error]                                                                                                 ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/SparkConf.java:187: error: ConfigEntry is not public in org.apache.spark.internal.config; cannot be accessed from outside package
[error]    <T extends java.lang.Object> org.apache.spark.SparkConf setIfMissing (org.apache.spark.internal.config.ConfigEntry<T> entry, T value)  { throw new RuntimeException(); }
[error]                                                                                                          ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/SparkConf.java:188: error: OptionalConfigEntry is not public in org.apache.spark.internal.config; cannot be accessed from outside package
[error]    <T extends java.lang.Object> org.apache.spark.SparkConf setIfMissing (org.apache.spark.internal.config.OptionalConfigEntry<T> entry, T value)  { throw new RuntimeException(); }
[error]                                                                                                          ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder/core/target/java/org/apache/spark/SparkConf.java:208: error: ConfigEntry is not public in org.apache.spark.internal.config; cannot be accessed from outside package
[error]     org.apache.spark.SparkConf remove (org.apache.spark.internal.config.ConfigEntry<?> entry)  { throw new RuntimeException(); }
```

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 2]
```diff
@@ -2657,22 +2661,27 @@ object functions {
...
 
+  // scalastyle:off line.size.limit
   /**
-   * Convert time string with given pattern
-   * (see [http://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html])
-   * to Unix time stamp (in seconds), return null if fail.
+   * Converts time string with given pattern to Unix timestamp (in seconds).
+   * Returns `null` if fails.
+   *
+   * @see [[http://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html Customizing Formats]]
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: It looks the documentation generation for Javadoc 8 is being failed due to `@see` with link - 

```
[error] /home/jenkins/workspace/SparkPullRequestBuilder/sql/core/target/java/org/apache/spark/sql/functions.java:2996: error: self-closing element not allowed
[error]    * @see <a href="http://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html Customizing Formats"/>
[error]           ^
[error] /home/jenkins/workspace/SparkPullRequestBuilder/sql/core/target/java/org/apache/spark/sql/functions.java:2996: error: invalid uri: "http://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html Customizing Formats"
[error]    * @see <a href="http://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html Customizing Formats"/>
[error]                                                                     ^
```

Probably, we should wrap it `href` as I did before - https://github.com/apache/spark/pull/16013 or find a way to make this link properly (or just remove it).

The other errors seem spurious. Please refer my observation - https://github.com/apache/spark/pull/17389#issuecomment-288438704

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 3]
```diff
@@ -2647,7 +2647,11 @@ object functions {
   }
 
   /**
-   * Gets current Unix timestamp in seconds.
+   * Returns the current Unix timestamp (in seconds).
+   *
+   * NOTE: All calls of `unix_timestamp` within the same query return the same value
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: Should we use `@note` and fix the corresponding contents in Python and R if applicable?

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 4]
```diff
@@ -2657,22 +2661,27 @@ object functions {
...
 
+  // scalastyle:off line.size.limit
   /**
-   * Convert time string with given pattern
-   * (see [http://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html])
-   * to Unix time stamp (in seconds), return null if fail.
+   * Converts time string with given pattern to Unix timestamp (in seconds).
+   * Returns `null` if fails.
+   *
+   * @see <a href="http://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html">Customizing Formats</a>
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]:
```
@see <a href="http://docs.oracle.com/javase/tutorial/i18n/format/simpleDateFormat.html">
Customizing Formats</a>
```

Given my past try, the above one works - https://github.com/apache/spark/pull/15999#discussion_r89456806 (IIRC, I checked the output).

**Review by HyukjinKwon — COMMENTED** [review]
These are all from me.

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala (hunk 4 above)
Comment (felixcheung) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: that can avoid having scalastyle:off

**Review by gatorsmile — APPROVED** [review]
LGTM

**Comment by gatorsmile** [comment]
Thanks! Merging to master.

**Pull request closed without merging**